### ISSUE TITLE
"This tracked scope should not be async." on onMount()

### ISSUE THREAD

**Issue by Brawl345** [joshwilsonvu/eslint-plugin-solid#7]
**Describe the bug**
Using "onMount()" with an async function throws:

```
ESLint: This tracked scope should not be async. Solid's reactivity only tracks synchronously.(solid/reactivity)
```

I used the example code from the official tutorial to verify: https://www.solidjs.com/tutorial/lifecycles_onmount?solved

**To Reproduce**
1. Go to https://www.solidjs.com/tutorial/lifecycles_onmount?solved
2. Copy the code and paste it in a file
3. Check it with ESLint

**Expected behavior**
No error should be shown because it still works.

**Screenshots**
![image](https://user-images.githubusercontent.com/5049892/150692186-7f5d1c50-7daa-4d99-83c7-22d2b142c10f.png)

**Desktop (please complete the following information):**
 - OS: Windows 8.1
 - Output of `node --version`: v16.13.2
 - Output of `npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`: `-- eslint-plugin-solid@0.4.1`
 - Output of `npm list eslint`/`yarn why eslint`:

```
+-- eslint-config-prettier@8.3.0
| `-- eslint@8.7.0 deduped
+-- eslint-plugin-solid@0.4.1
| +-- @typescript-eslint/experimental-utils@5.10.0
| | +-- @typescript-eslint/utils@5.10.0
| | | `-- eslint@8.7.0 deduped
| | `-- eslint@8.7.0 deduped
| `-- eslint@8.7.0 deduped
+-- eslint-plugin-unicorn@40.1.0
| +-- eslint-utils@3.0.0
| | `-- eslint@8.7.0 deduped
| `-- eslint@8.7.0 deduped
`-- eslint@8.7.0

```

**Additional context**
N/A

**Comment by joshwilsonvu** [comment]
Thanks for the report and for trying out this plugin. This is a bug, and once I have it fixed I'll add this code to the tests.

I think I'm going to change the implementation so it only warns if a signal is actually used past an `await`, with exceptions for `onMount` and a few other functions. That should cover it, I'll update you when the fix is released. 

For now I hope disabling the rule for each line this affects won't be too much of an inconvenience.

**Comment by Brawl345** [comment]
Thanks, take your time, no problem :)

**Comment by joshwilsonvu** [comment]
Got to it tonight, should be fixed in `v0.4.3`. Pretty simple change. Thanks again!

**Comment by Brawl345** [comment]
Works, thanks! :+1:
